app/src/utils/spatial-utils.tsx

```diff
@@ -0,0 +1,194 @@
+import { IMarker, IMarkerLayer } from 'components/map/components/MarkerCluster';
+import { IStaticLayer } from 'components/map/components/StaticLayers';
+import DatasetPopup from 'components/map/DatasetPopup';
+import FeaturePopup, { BoundaryCentroidFeature, BoundaryFeature, OccurrenceFeature } from 'components/map/FeaturePopup';
+import { LAYER_NAME, SPATIAL_COMPONENT_TYPE } from 'constants/spatial';
+import { Feature } from 'geojson';
+import { EmptyObject, ISpatialData, ITaxaData } from 'interfaces/useObservationApi.interface';
+import { LatLngTuple } from 'leaflet';
+import { isObject } from 'lodash-es';
+import React from 'react';
+
+export interface IDataResult {
+  key: string;
+  name: string;
+  count: number;
+}
+
+export interface ISpatialDataGroupedBySpecies {
+  [species: string]: ISpatialData[];
+}
+
+export const parseSpatialDataByType = (spatialDataRecords: ISpatialData[]) => {
+  const occurrencesMarkerLayer: IMarkerLayer = { layerName: LAYER_NAME.OCCURRENCES, markers: [] };
+  const boundaryStaticLayer: IStaticLayer = { layerName: LAYER_NAME.BOUNDARIES, features: [] };
+
+  for (const spatialRecord of spatialDataRecords) {
+    for (const feature of spatialRecord.spatial_data.features) {
+      const visible = true;
```

Comment:
Could we define this outside the scope of the nested `for` loops for efficiency sake? Perhaps it could even be removed (see next comment)